File: api/create_file_surgery_paths.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';
import * as dotenv from 'dotenv';
import { MongoClient } from 'mongodb';

interface FileSurgeryPaths {
	current: string;
	new: string;
	redirect?: string;
	name?: string;
}

export default (request: VercelRequest, response: VercelResponse) => {
	dotenv.config();
	const client = new MongoClient(process.env.MONGODB_URI || "");
	client.connect().then(() => {
		const collection = client.db("data").collection("file_surgery_paths");
		try {
			let body = request.body;
			let tmp: FileSurgeryPaths[] = []
			for (let i = 0; i < body.data.length; i++) {
				let data: FileSurgeryPaths = {
					"current": body.data[i]["current"],  // required current path
					"new": body.data[i]["new"],  // required new path
				}
				if ("redirect" in body.data[i]) {
					data.redirect = body.data[i]["redirect"]  // optional redirect
				}
				if ("name" in body.data[i]) {
					data.name = body.data[i]["name"];  // optional name
				}
				tmp.push(data);
			}
			collection.insertMany(tmp);
			response.status(200).send(request.body);
		} catch (err) {
			response.status(400).json({ error: "bad body format" });
		}
	}).catch((err) => {
		response.status(500).json({ error: "could not connect to database" });
	});
	client.close();
};

File: api/create_file_surgery_paths.py
from http.server import BaseHTTPRequestHandler
import os
from pymongo import MongoClient
import json
from flask import Flask, Response
app = Flask(__name__)

@app.route('/', defaults={'path': ''}, methods=['PUTS'])
@app.route('/<path:path>', methods=['PUTS'])
def catch_all(path):
    from flask import request
    if request.headers.get('Content-Type') != 'application/json': return Response("Content-Type must be application/json", mimetype='text/plain', status=400)
    body = request.json
    data = [{
        "name": i["name"] if "name" in i.keys() else None,
        "new": i["new"],
        "curret": i["current"],
        "redirect": i["redirect"] if "redirect" in i.keys() else None
    } for i in body["data"]]
    client = MongoClient(os.environ['MONGODB_URI'])
    client.data.file_surgery_paths.insert_many(data)
    return Response("OK", mimetype='text/plain', status=200)
